src/Services/CustomerOptionValueArchiver.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brille24\SyliusCustomerOptionsPlugin\Services;
+
+use Brille24\SyliusCustomerOptionsPlugin\Entity\OrderItemInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Order\Model\OrderInterface;
+use Webmozart\Assert\Assert;
+
+final class CustomerOptionValueArchiver implements OrderItemValueCopierInterface
+{
+    /** @var ChannelInterface */
+    private $currentChannel;
+
+    public function __construct(ChannelInterface $currentChannel)
+    {
+        $this->currentChannel = $currentChannel;
+    }
+
+    /** {@inheritdoc} */
+    public function copyOverValues(OrderInterface $order): void
+    {
+        foreach ($order->getItems() as $orderItem) {
+            if (!$orderItem instanceof OrderItemInterface) {
+                continue;
+            }
+
+            $this->copyOverValuesForOrderItem($orderItem);
+        }
+    }
+
+    /** {@inheritdoc} */
+    public function copyOverValuesForOrderItem(OrderItemInterface $orderItem): void
```

Comment:
Update documentation on when this happens and why this happens as just talked about.

Reply:
Documentation can be found at docs/OrderProcessor_Usage.png

Reply:
Can you add a reference to that file here please? :slightly_smiling_face:

Reply:
Done